05/app.js

```diff
@@ -0,0 +1,64 @@
+function Student( firstName , secondName ) {
+
+    this.firstName = firstName,
+    this.secondName = secondName,
+
+    this.grades = {};
+}
+
+
+Student.prototype.addGrade = function(subject, grades) {
+
+
+      if(typeof this.grades[subject] === 'undefined') {
+          this.grades[subject] = [];
+          this.grades[subject].push(grades);
+
+      } else {
+          this.grades[subject].push(grades);
```

Comment:
Zobacz, że te dwa elementy się powtarzają. Można to zrobić "ładniej":
```
if(typeof this.grades[subject] === 'undefined') {
      this.grades[subject] = [];
} 

this.grades[subject].push(grades);
```

Reply:
Przy zapisie, który proponujesz przy wywołaniu metody np. student.addGrade('polski, 5'), doda nam tylko przedmiot 'polski' z pustą tablicą, Specjalnie zamieściłem ten zapis, żeby przy wykryciu braku danego przedmiotu dodał go razem z oceną.

Reply:
Nie prawda. Sprawdziłeś to? Jeśli nie ma jeszcze żadnej oceny to doda tam pustą tablicę i potem do niej doda ocenę.

Reply:
Wyjaśnione ! :D